Framework/PythonInterface/mantid/utils/absorptioncorrutils.py

```diff
@@ -32,174 +33,245 @@ def _getBasename(filename):
     elif abs_method == "SampleAndContainer":
-        wsn_as = f"{absName}_ass"
-        wsn_ac = f"{absName}_acc"
+        abs_wksp_sample = f"{prefix_name}_ass"
+        abs_wksp_container = f"{prefix_name}_acc"
     elif abs_method == "FullPaalmanPings":
-        wsn_as = f"{absName}_assc"
-        wsn_ac = f"{absName}_ac"
+        abs_wksp_sample = f"{prefix_name}_assc"
+        abs_wksp_container = f"{prefix_name}_ac"
     else:
         raise ValueError("Unrecognized absorption correction method '{}'".format(abs_method))
 
-    # step_1: check memory to see if the ws is already there
-    # -- check SHA1
-    # NOTE: The new attribute absSHA1 should be there.
-    if mtd.doesExist(wsn_as):
-        found_as = mtd[wsn_as].run()["absSHA1"].value == sha1
-    if mtd.doesExist(wsn_ac):
-        found_ac = mtd[wsn_ac].run()["absSHA1"].value == sha1
-
-    # step_2: try to load from cache_file provided
-    if (not found_as) or (not found_ac):
-        # disk -> memory
-        if os.path.exists(cache_file_name):
-            # LoadNexus(Filename=cache_file_name, OutputWorkspace=wsn_as)
-            # if mtd[wsn_as].id() == "WorkspaceGroup":
-            #     UnGroupWorkspace(InputWorkspace=wsn_as)
-            wsntmp = f"{wsn_as}_wsg"
-            Load(Filename=cache_file_name, OutputWorkspace=wsntmp)
-            wstype = mtd[wsntmp].id()
-            if wstype == "Workspace2D":
-                RenameWorkspace(InputWorkspace=wsntmp, OutputWorkspace=wsn_as)
-            elif wstype == "WorkspaceGroup":
-                # there should be exactly two workspaces inside, ungroup them will
-                # restore them with the orignal name (wsn_as, wsn_ac)
-                UnGroupWorkspace(InputWorkspace=wsntmp)
+    # step_1: check memory
+    if mtd.doesExist(abs_wksp_sample):
+        found_abs_wksp_sample = mtd[abs_wksp_sample].run()["absSHA1"].value == sha1
+    if mtd.doesExist(abs_wksp_container):
+        found_abs_wksp_container = mtd[abs_wksp_container].run()["absSHA1"].value == sha1
+
+    # step_2: load from disk if either is not found in memory
+    cache_file_name = ""
+    if (not found_abs_wksp_sample) or (not found_abs_wksp_container):
+        for candidate in cache_files :
+            if os.path.exists(candidate):
+                cache_file_name = candidate
+                wsntmp = f"tmpwsg"
+                Load(Filename=cache_file_name, OutputWorkspace=wsntmp)
+                wstype = mtd[wsntmp].id()
+                if wstype == "Workspace2D":
+                    RenameWorkspace(InputWorkspace=wsntmp, OutputWorkspace=abs_wksp_sample)
+                elif wstype == "WorkspaceGroup":
+                    UnGroupWorkspace(InputWorkspace=wsntmp)
+                else:
+                    raise ValueError(f"Unsupported cached workspace type: {wstype}")
+                break
+
+        if( cache_files and not cache_file_name ):
+            raise( RuntimeError("No valid cache files specified") )
```

Comment:
Implicit casting to bool always confuses me.
Can we switch to something slightly easier to understand?
For instance, we could directly use `candidate` for the loading part (L144), then assign `candidate` to `cache_file_name` right before break where we are confident that workspaces are loaded from file.
Then the remaining if-block at L154 can be change into 
```python
if cache_file_name=="":
    raise( RuntimeError("No valid cache files specified") )
```